### PR TITLE
state min HHVM version

### PR DESCRIPTION
**Pull request by keradus**
remove HHVM hacks after fixes of
- https://github.com/facebook/hhvm/issues/3703
- https://github.com/facebook/hhvm/issues/3629

probably need to wait a bit for stable HHVM release

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Maybe we should still include this work around, but just do a comparison on the hhvm version?

**Comment by GrahamCampbell** [comment]
Looks like it'll be ages until this gets into the next hhvm lts release. We'll have to wait until 3.9 is released for that.

**Comment by keradus** [comment]
No need to merge it now, as I said we can wait a bit.

I hate any hacks for hhvm bugs, when they fix it we will remove it. We do not support multiple hhvm versions.

**Comment by GrahamCampbell** [comment]
> No need to merge it now, as I said we can wait a bit.

I'm suggesting we modify it to instead do a comparison of the hhvm version. It would be incorrect for us to apply this patch to hhvm 3.7+.

**Comment by keradus** [comment]
Yes, I do understand you, just not agree with it.
We do not support hhvm like 3.6, 3.7 etc. Just the latest stable.
This is not our bug that we should fix - it's hhvm's bug that they should fix. And they do.
We should not create a workaround for every bug that hhvm has have.

We are already super cool that we have removed hhvm from our allow_failures matrix ;)

**Comment by gharlan** [comment]
I'm with @keradus, supporting the latest stable hhvm is enough.

**Comment by stof** [comment]
@keradus HHVM 3.3 is an LTS version, so it should be supported too

**Comment by GrahamCampbell** [comment]
> HHVM 3.3 is an LTS version, so it should be supported too

That's what I said. Also, so is HHVM 3.6, and only HHVM 3.7+ has the fix.

**Comment by GrahamCampbell** [comment]
Or, rather, 3.7+ has all the fixes, earlier versions had other stuff fixed.

**Comment by keradus** [comment]
> @keradus HHVM 3.3 is an LTS version, so it should be supported too

then the bug fixes should go there, don't they?

**Comment by stof** [comment]
> then the bug fixes should go there, don't they?

They don't backport all bug fixes, no. See http://hhvm.com/blog/6083/hhvm-long-term-support for their description of what goes in the LTS.

**Comment by GrahamCampbell** [comment]
> then the bug fixes should go there, don't they?

No, only security fixes.

**Comment by keradus** [comment]
I really do not want to support bunch of hhvm version.
Personally I would throw hhvm support away, but I won't, due to fact some of us want to use it. It was hard enough to remove hhvm from our failure matrix.
Fb had already shown few times that they don't care to be compatible with PHP's tokenizer.
Therefore, IMO, is is enough to keep support of latest stable.
We could even terminate tool on unsupported version like we do with old PHP.

**Comment by GrahamCampbell** [comment]
> I really do not want to support bunch of hhvm version.

I agree.

> Personally I would throw hhvm support away

I think we should still try to support HHVM though, even if we support the latest versions only.

**Comment by stof** [comment]
> We could even terminate tool on unsupported version like we do with old PHP.

if we terminate clearly (with an explanation), I agree about such change

**Comment by keradus** [comment]
Then decided.
To be honest I don't know how to properly detect version of hhvm - anyone willing to create such PR ?

**Comment by stof** [comment]
@keradus they have a ``HHVM_VERSION`` constant. You already use the fact that the constant is defined to detect HHVM. Just use the value of the constant to get the version :smile:

**Comment by keradus** [comment]
HHVM version check added

File: Symfony/CS/Resources/phar-stub.php [hunk 1]
```diff
@@ -10,7 +10,12 @@
  * with this source code in the file LICENSE.
  */
 
-if (version_compare(phpversion(), '5.3.6', '<')) {
+if (defined('HHVM_VERSION_ID')) {
+    if (HHVM_VERSION_ID < 30500) {
+        fwrite(STDERR, "HHVM needs to be a minimum version of PHP 3.5.0\n");
```
Comment (gharlan) [inline Symfony/CS/Resources/phar-stub.php]: PHP 3.5.0?

**Comment by keradus** [comment]
btw, this started to work on hhvm-nightly ;)

File: Symfony/CS/Resources/phar-stub.php (hunk 1 above)
Comment (keradus) [inline Symfony/CS/Resources/phar-stub.php]: HHVM 3.5.0

Comment (keradus) [inline Symfony/CS/Resources/phar-stub.php]: done

**Comment by stof** [comment]
@keradus looks like 3.5.0 is not actually the lowest version for which the current hacks are not necessary anymore

**Comment by keradus** [comment]
Yeah, we will need to bump it when current dev will become current stable, just don't know what the number will be.
One way or another it won't be merged until then, so we just need to wait a bit longer.

**Comment by GrahamCampbell** [comment]
> Yeah, we will need to bump it when current dev will become current stable, just don't know what the number will be.

Yes we do. The master hhvm branch will be the next minor version, guarenteed.

**Comment by keradus** [comment]
good to go

**Comment by GrahamCampbell** [comment]
:+1:

**Pull request closed without merging**